stan/math/prim/arr/functor/mpi_cluster.hpp

```diff
@@ -175,24 +174,29 @@ struct mpi_cluster {
    * mpi_stop_worker command.
    */
   void stop_listen() {
-    if (rank_ == 0 && mpi_cluster::cluster_listens_) {
+    if (rank_ == 0 && listening_status()) {
       mpi_broadcast_command<mpi_stop_worker>();
     }
-    mpi_cluster::cluster_listens_ = false;
+    listening_status() = false;
   }
 
   /**
    * Returns the current listening state of the cluster.
    */
-  static bool is_listening() { return cluster_listens_; }
+  static bool& listening_status() {
```

Comment:
I think we probably want to just pass this [primitive by reference](https://stackoverflow.com/a/14013189/212348) (cc @bob-carpenter )

Reply:
I am not sure what you mean here. What I am doing is the safest way to have the listening variable global and have it writable at the same time.

(I need to refactor this mpi_cluster class to become a Meyer-Singleton ... after this first release)

Reply:
The reason to pass by reference is if you want to be able to set it as:

```
listening_status() = false;
```

or perhaps more understandably,

```
bool& a = listening_status();
a = false;
```

Otherwise, return by value.

Reply:
Oops, I meant to say by value in my original post (the link I linked to said this but my bad). That's probably at least half of why it was confusing, haha.

I missed that you were setting it! My bad. I think I like Bob's suggestion for 
```
bool& a = listening_status();
a = false;
```
If you don't mind